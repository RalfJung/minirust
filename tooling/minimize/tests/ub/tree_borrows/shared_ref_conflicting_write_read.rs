//@ compile-flags: --minimize-tree-borrows
fn main() {
    let x = &mut 31; // (x, Reserved)   
    let xraw = x as *mut i32; // (x, Reserved) 
    let y = unsafe { &*xraw }; // (x, Reserved) -> (y, Frozen)
    
    unsafe { *xraw = 42; } // (x, Active) -> (y, Disable)

    assert!(*y == 42); // UB! Child Read to Disabled
}
