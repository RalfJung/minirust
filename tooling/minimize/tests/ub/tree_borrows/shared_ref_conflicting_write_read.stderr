fatal error: UB: Tree Borrows: Child reading a pointer without permission
